drivers/ethernet/eth_nxp_enet.c

```diff
@@ -387,22 +404,21 @@ static int eth_nxp_enet_rx(const struct device *dev)
 	return -EIO;
 }
 
-static void eth_nxp_enet_rx_thread(void *arg1, void *unused1, void *unused2)
+static void eth_nxp_enet_rx_thread(struct k_work *work)
 {
-	const struct device *dev = arg1;
+	struct nxp_enet_mac_data *data =
+		CONTAINER_OF(work, struct nxp_enet_mac_data, rx_work);
+	const struct device *dev = data->dev;
 	const struct nxp_enet_mac_config *config = dev->config;
-	struct nxp_enet_mac_data *data = dev->data;
 
-	while (1) {
-		if (k_sem_take(&data->rx_thread_sem, K_FOREVER) == 0) {
-			while (eth_nxp_enet_rx(dev) == 1) {
-				;
-			}
-			/* enable the IRQ for RX */
-			ENET_EnableInterrupts(config->base,
-			  kENET_RxFrameInterrupt | kENET_RxBufferInterrupt);
-		}
+	if (k_sem_take(&data->rx_thread_sem, K_FOREVER)) {
+		return;
 	}
+
+	while (eth_nxp_enet_rx(dev) == 1)
+		;
```

Comment:
missing `{}` although this looks just fine too. I think there was some Misra rule for this, and a compliance checker script might complain about the missing braces.

Unrelated to this PR, I noticed while checking what `eth_nxp_enet_rx()` is doing, that it might make sense in that function to return the real error from `net_recv_data()` instead of always `-EIO`

Reply:
WAIT_FOR didn't make sense to me to use because, we aren't waiting on something, ie there is no notion of time here, it's just repeat this routine until done, so I elected to convert it to a do-while with a return variable